plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteCaseSensitiveComparison.java

```diff
@@ -0,0 +1,94 @@
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argument;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argumentCount;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.call;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.functionName;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.plugin.jdbc.CaseSensitivity.CASE_SENSITIVE;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+
+public class RewriteCaseSensitiveComparison
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> LEFT = newCapture();
+    private static final Capture<Variable> RIGHT = newCapture();
+
+    private final Pattern<Call> pattern;
+
+    public RewriteCaseSensitiveComparison(Set<ComparisonOperator> enabledOperators)
+    {
+        Set<FunctionName> functionNames = enabledOperators.stream()
+                .map(ComparisonOperator::getFunctionName)
+                .collect(toImmutableSet());
+
+        pattern = call()
+                .with(type().equalTo(BOOLEAN))
+                .with(functionName().matching(functionNames::contains))
+                .with(argumentCount().equalTo(2))
+                .with(argument(0).matching(variable().with(type().matching(VarcharType.class::isInstance)).capturedAs(LEFT)))
+                .with(argument(1).matching(variable().with(type().matching(VarcharType.class::isInstance)).capturedAs(RIGHT)));
+    }
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return pattern;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        ComparisonOperator comparison = ComparisonOperator.forFunctionName(expression.getFunctionName());
+        Variable firstArgument = captures.get(LEFT);
+        Variable secondArgument = captures.get(RIGHT);
+
+        if (!isCaseSensitive(firstArgument, context) || !isCaseSensitive(secondArgument, context)) {
+            return Optional.empty();
+        }
+        return context.defaultRewrite(firstArgument).flatMap(first ->
+                context.defaultRewrite(secondArgument).map(second ->
+                        new ParameterizedExpression(
+                                "(%s) %s (%s)".formatted(first.expression(), comparison.getOperator(), second.expression()),
+                                ImmutableList.<QueryParameter>builder()
+                                        .addAll(first.parameters())
+                                        .addAll(second.parameters())
+                                        .build())));
+    }
+
+    private static boolean isCaseSensitive(Variable variable, RewriteContext<?> context)
```

Comment:
nit: move RewriteContext  as first argument

Reply:
i put it last, because it's last in `io.trino.plugin.base.expression.ConnectorExpressionRule#rewrite` (the interface)